src/mm/model_allocator.c

```diff
@@ -0,0 +1,216 @@
+/**
+ * @file mm/model_allocator.c
+ *
+ * @brief The subsystem managing the memory state of an LP
+ *
+ * SPDX-FileCopyrightText: 2008-2025 HPCS Group <rootsim@googlegroups.com>
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+#include <errno.h>
```

Comment:
Let's have a consistent order of header inclusions:
1. The header related to the source file followed by a new line (in this case `model_allocator.h`
2. Headers that belong to this project followed by a new line
3. System headers